src/neurotechdevkit/results/_results.py

```diff
@@ -420,6 +424,182 @@ def render_steady_state_amplitudes_3d(
         """
         rendering.render_amplitudes_3d_with_napari(self)
 
+    def _validate_slice_args(
+        self, slice_axis: int | None, slice_position: float | None
+    ) -> None:
+        """Validate that slicing axis and position are within scenario range.
+
+        `slice_axis` should be either 0, 1, or 2 (for X, Y, Z).
+        `slice_position` must be within boundaries for `slice_axis` extent.
+
+        Args:
+            slice_axis: the axis along which to slice the 3D field to be recorded. If
+                None, then the complete field will be recorded. Use 0 for X axis, 1
+                for Y axis and 2 for Z axis.
+            slice_position: the position (in meters) along the slice axis at
+                which the slice of the 3D field should be made.
+
+        Raises:
+            ValueError if axis is not 0, 1, 2.
+            ValueError if `slice_position` falls outside the current range of
+                `slice_axis`.
+            ValueError if  `slice_axis` is None but `slice_position` is not None and
+                vice versa.
+        """
+        if slice_axis not in (0, 1, 2):
```

Comment:
Good catch, I stole this function from `_base.py` and didn't notice that we get the wrong value error if slice_axis is None. With your suggested change, if both arguments are None then we run into an error. Both here and in base, before the function is called, it is made sure that `slice_axis` and `slice_position` are not both None, but the linter is still worried that it can be none. I instead changed it to not allow either to be None in this function and also made this same change to the instance of this function in `_base.py`. I didn't work with the instance in base, so it would be great if you could check that it does make sense to make the change there as well.

Reply:
Got it - yeah this update makes sense to me! sometimes the linter is not as smart as we'd like.

BTW, is this `SteadyStateResult3D._validate_slice_args` version used anywhere? If not, you could take it out and just keep the `Scenario3D._validate_slice_args` version you're using down below: less code to maintain in the future.

Reply:
That makes sense, removed the `SteadyStateResult3D` version.

Reply:
👍🏼 Looks good to me to merge.